testsuite/forge/src/backend/k8s/cluster_helper.rs

```diff
@@ -826,15 +826,41 @@ fn dump_helm_values_to_file(helm_release_name: &str, tmp_dir: &TempDir) -> Resul
 
 #[derive(Error, Debug)]
 #[error("{0}")]
-enum ApiError {
+pub enum ApiError {
     RetryableError(String),
     FinalError(String),
 }
 
-async fn create_namespace(
+/// Does the same as create_namespace and handling the 409, but for any k8s resource T
+pub async fn maybe_create_k8s_resource<T>(
```

Comment:
We can actually use Dynamic object here and a nice little macro to get the api that you want like so

```
macro_rules! kube_api {
    ($client:expr, $t:ty) => {{
        let tm = kube::api::TypeMeta::resource::<$t>();
        let gvk = kube::api::GroupVersionKind::try_from(&tm).unwrap();
        let ar = kube::api::ApiResource::from_gvk(&gvk);

        kube::api::Api::<kube::api::DynamicObject>::all_with($client.clone(), &ar)
    }};
    ($client:expr, $t:ty, $namespace:expr) => {{
        let tm = kube::api::TypeMeta::resource::<$t>();
        let gvk = kube::api::GroupVersionKind::try_from(&tm).unwrap();
        let ar = kube::api::ApiResource::from_gvk(&gvk);

        kube::api::Api::<kube::api::DynamicObject>::namespaced_with(
            $client.clone(),
            &$namespace,
            &ar,
        )
    }};
}
```